src/tree.rs

```diff
@@ -2204,3 +2204,10 @@ impl fmt::Display for CompareAndSwapError {
         write!(f, "Compare and swap conflict")
     }
 }
+
+impl std::error::Error for CompareAndSwapError {
+    #[allow(deprecated, deprecated_in_future)]
+    fn description(&self) -> &str {
+        "Compare and swap conflict"
+    }
+}
```

Comment:
As I read [the description of the Error trait](https://doc.rust-lang.org/std/error/trait.Error.html#method.description), it is better to avoid using this method, and impl fmt::Dsiplay on the error type.

Reply:
Ahh I wasn't aware, in that case I'll close this. Thanks!

Reply:
Err, misunderstood, just removed `description`s and will just avoid that method.